Purchases/Misc/Obsoletions.swift

```diff
@@ -174,17 +174,55 @@ public extension Purchases {
      * If the purchase was not successful, there will be an `NSError`.
      * If the user cancelled, `userCancelled` will be `YES`.
      */
-    @available(iOS, obsoleted: 1, renamed: "purchase(product:discount:completion:)")
-    @available(tvOS, obsoleted: 1, renamed: "purchase(product:discount:completion:)")
-    @available(watchOS, obsoleted: 1, renamed: "purchase(product:discount:completion:)")
-    @available(macOS, obsoleted: 1, renamed: "purchase(product:discount:completion:)")
-    @available(iOS 12.2, macOS 10.14.4, macCatalyst 13.0, tvOS 12.2, watchOS 6.2, *)
+    @available(iOS, introduced: 12.2, obsoleted: 12.2, renamed: "purchase(product:discount:completion:)")
+    @available(tvOS, introduced: 12.2, obsoleted: 12.2, renamed: "purchase(product:discount:completion:)")
+    @available(watchOS, introduced: 6.2, obsoleted: 6.2, renamed: "purchase(product:discount:completion:)")
+    @available(macOS, introduced: 10.14.4, obsoleted: 10.14.4, renamed: "purchase(product:discount:completion:)")
+    @available(macCatalyst, introduced: 13.0, obsoleted: 13.0, renamed: "purchase(product:discount:completion:)")
     func purchaseProduct(_ product: SKProduct,
                          discount: SKPaymentDiscount,
                          _ completion: @escaping PurchaseCompletedBlock) {
         fatalError()
     }
 
+    /**
+     * Use this function to retrieve the `SKPaymentDiscount` for a given `SKProduct`.
+     *
+     * - Parameter discount: The `SKProductDiscount` to apply to the product.
+     * - Parameter product: The `SKProduct` the user intends to purchase.
+     * - Parameter completion: A completion block that is called when the `SKPaymentDiscount` is returned.
+     * If it was not successful, there will be an `Error`.
+     */
+    @available(iOS, introduced: 12.2, obsoleted: 12.2, message: "Obtain StoreProductDiscount from StoreProduct")
+    @available(tvOS, introduced: 12.2, obsoleted: 12.2, message: "Obtain StoreProductDiscount from StoreProduct")
+    @available(watchOS, introduced: 6.2, obsoleted: 6.2, message: "Obtain StoreProductDiscount from StoreProduct")
+    @available(macOS, introduced: 10.14.4, obsoleted: 10.14.4, message: "Obtain StoreProductDiscount from StoreProduct")
+    @available(macCatalyst, introduced: 13.0, obsoleted: 13.0, message: "Obtain StoreProductDiscount from StoreProduct")
+    @objc(paymentDiscountForProductDiscount:product:completion:)
+    func paymentDiscount(forProductDiscount discount: SKProductDiscount,
```

Comment:
Glad you caught this! I'm realizing now that we updated the type to `StoreProductDiscount`, but really we want to deprecate the old method, which took in a `SKProductDiscount`

Reply:
I was reading the API changes of https://github.com/RevenueCat/purchases-ios/pull/1208 and got curious about one thing. If there's no `Purchases.paymentDiscount(forProductDiscount:product:completion:)` API anymore, and there's just a `purchase(package:discount:)` that accepts a product discount, how do developers determine a customer is capable of receiving a discount? 

The reason it was separated in two APIs, is that the first one, the one that gets the SKPaymentDiscount and is being obsoleted with this change, can be used to determine if a discount can be offered. I am not sure how devs can do that anymore.